### PR TITLE
blob/s3blob: support v1 ListObjects

### PR DESCRIPTION
**Pull request by EvyBongers**
Be compatible with CEPH as blob store provider

See https://tracker.ceph.com/issues/37801 for the bug report with the CEPH project.

### CODE REVIEW

File: blob/s3blob/s3blob.go [hunk 1]
```diff
@@ -134,19 +134,25 @@ func (o *URLOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.Bucket
 }
 
 // Options sets options for constructing a *blob.Bucket backed by fileblob.
-type Options struct{}
+type Options struct {
+	UseLegacyListObjects bool
```
Comment (vangent) [inline blob/s3blob/s3blob.go]: Please add a docstring for the new option. Perhaps explain that some providers have S3-compatible APIs, but don't support the ListObjectV2 API yet.

File: blob/s3blob/s3blob.go [hunk 2]
```diff
@@ -366,6 +377,89 @@ func (b *bucket) ListPaged(ctx context.Context, opts *driver.ListOptions) (*driv
 	return &page, nil
 }
 
+func (b *bucket) legacyListPaged(ctx context.Context, opts *driver.ListOptions) (*driver.ListPage, error) {
+	pageSize := opts.PageSize
```
Comment (vangent) [inline blob/s3blob/s3blob.go]: It looks like a high percentage of the logic is shared between `ListPaged` and `legacyListPaged`. Could you try sharing more of it? For example, you could make only the `ListObjectsRequest` call conditional, and in the legacy version convert from the V2 objects to the V1 objects (and from V1 to V2 in the result).

File: blob/s3blob/s3blob.go (hunk 1 above)
Comment (vangent) [inline blob/s3blob/s3blob.go]: If I understand correctly, S3 itself should support both modes, right?

If so, I'd like to make the tests for S3 run in both modes so we have coverage of both paths. I could do this as a follow-up on PR from yours.

Comment (vangent) [reply blob/s3blob/s3blob.go]: I verified that this works (i.e., I ran all of the conformance tests for S3 using `UseLegacyListObjects=true` and they passed). So, resolving this comment.

Comment (vangent) [inline blob/s3blob/s3blob.go]: Maybe just `UseLegacyList` ?

File: blob/s3blob/s3blob.go (hunk 2 above)
Comment (EvyBongers) [reply blob/s3blob/s3blob.go]: Hi @vangent,

I've updated the code (see 24d73e47894c24b023fc082c5bdc8290d3c4919b). To me the previous solution looks cleaner. What do you think?

Comment (vangent) [reply blob/s3blob/s3blob.go]: I like this one, thanks!

**Pull request merged**